physical/raft/raft.go

```diff
@@ -274,7 +274,7 @@ func EnsurePath(path string, dir bool) error {
 	if !dir {
 		path = filepath.Dir(path)
 	}
-	return os.MkdirAll(path, 0o755)
+	return os.MkdirAll(path, 0o750)
```

Comment:
Is there a reason why this is 0o750 instead of 0o700?

Reply:
I ran this tool  https://github.com/securego/gosec#github-action and the maximum allowed permission that it suggested was 0o750. So, I went with that. Would you think that 0o700 is better?

Reply:
I think my question to follow up the original question would be is there a reason that we want to allow group read and execute access? It might be safest to only allow owner access but I be missing some context.

Reply:
These are permissions for a directory, and this would allow non-vault user to traverse and list the directory contents. While using `0o750` should only allow the disclose the filenames in the directory (which are likely not sensitive), it might make sense to remain consistent with other folder permissions, and remove group read/execute bits, as mentioned above (similar to https://github.com/hashicorp/vault/blob/main/sdk/physical/file/file.go#L239)

Reply:
I guess 0o700 follows the least privilege rule the best. Done